pkg/agent/config/config.go

```diff
@@ -429,6 +431,18 @@ func get(envInfo *cmds.Agent, proxy proxy.Proxy) (*config.Node, error) {
 	nodeConfig.CACerts = info.CACerts
 	nodeConfig.Containerd.Config = filepath.Join(envInfo.DataDir, "agent", "etc", "containerd", "config.toml")
 	nodeConfig.Containerd.Root = filepath.Join(envInfo.DataDir, "agent", "containerd")
+	switch nodeConfig.AgentConfig.Snapshotter {
+	case "overlayfs":
```

Comment:
Should we have a default case to catch anything that doesn't match the preceding cases and report a warning or error?

Reply:
It's pluggable - there are a variety of in-tree snapshotters and more can be loaded at runtime, so I don't think we need to handle all possible values.
https://github.com/containerd/containerd/tree/master/snapshots

Reply:
Sure, but what should the expected behavior be if one is entered that doesn't exist or has a typo in the name?

Reply:
Containerd currently fails to start up if you request a snapshotter that doesn't exist, for example if you run `k3s server --snapshotter INVALID`:

```
time="2020-12-01T09:40:23.106999377-08:00" level=warning msg="failed to load plugin io.containerd.grpc.v1.cri" error="failed to create CRI service: failed to find snapshotter \"INVALID\""
```

I don't think we need to fix that here though, do we? If we need to do additional validation of containerd configuration lets reserve that for a separate PR.

Reply:
As long as it's not ignored or a silent failure, that's cool and since Containerd takes care of it, that works for me.